server/link.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// Link represents a pattern to autolink.
+type Link struct {
+	Name                 string
+	Disabled             bool
+	Pattern              string
+	Template             string
+	Scope                []string
+	WordMatch            bool
+	DisableNonWordPrefix bool
+	DisableNonWordSuffix bool
+
+	template  string
+	re        *regexp.Regexp
+	multipass bool
+}
+
+// DisplayName returns a display name for the link.
+func (l Link) DisplayName() string {
+	if l.Name != "" {
+		return l.Name
+	}
+	return l.Pattern
+}
+
+// Compile compiles the link's regular expression
+func (l *Link) Compile() error {
+	if l.Disabled || len(l.Pattern) == 0 || len(l.Template) == 0 {
+		return nil
+	}
+
+	pattern := l.Pattern
+	template := l.Template
+	multipass := false
```

Comment:
This (`multipass`) probably warrants a comment. Previously, word boundaries were defined as custom regexp's. These patterns consume characters, so they can not be used with `ReplaceAll` functions. Example, if the pattern is defined as `"X"`, `"foo X bar"` the X will match, but in `"foo X X bar"` only the first X matches, the match consumes the subsequent space, and then the second X doesn't match.

see the `Replace()` method for more.

Reply:
With better naming, didn't need to comment as much